opal/mca/reachable/netlink/reachable_netlink_module.c

```diff
@@ -13,12 +15,27 @@
 #include "opal/constants.h"
 #include "opal/types.h"
 
+#ifdef HAVE_MATH_H
+#include <math.h>
+#endif
+
+#include "opal/util/net.h"
 #include "opal/mca/reachable/base/base.h"
 #include "reachable_netlink.h"
+#include "libnl_utils.h"
+
+enum connection_quality {
+    CQ_NO_CONNECTION = 0,
+    CQ_DIFFERENT_NETWORK = 50,
+    CQ_SAME_NETWORK = 100
+};
```

Comment:
You define the same values in `reachable_weighted.c` (and also the TCP BTL).  Should these be put in `reachable.h` and prefixed with `OPAL_REACHABILITY_` or somesuch?

Reply:
I thought about it, but I don't actually want them to be part of the public interface; it means we can't tweak the algorithms and have to live with those choices going forward.  Imagine a netlink version that counted router hops as part of the weight.  Or ping latency, or something like that.  Those would be hard if we baked the constants into the API.

If you look at the usNIC BTL (which is where we should be moving the TCP BTL), there's no need for these constants, because it's the relative weights that matter.  So in time, we'll be removing them from the TCP BTL.